actions/apps.ts

```diff
@@ -15,16 +15,81 @@ import AppsForm from 'components/apps_form';
 import {ModalIdentifiers} from 'utils/constants';
 import {getSiteURL, shouldOpenInNewTab} from 'utils/url';
 import {browserHistory} from 'utils/browser_history';
-import {makeCallErrorResponse} from 'utils/apps';
+import {createCallRequest, makeCallErrorResponse} from 'utils/apps';
 
 import {cleanForm} from 'mattermost-redux/utils/apps';
 
 import {sendEphemeralPost} from './global_actions';
 
-export function doAppCall<Res=unknown>(call: AppCallRequest, type: AppCallType, intl: any): ActionFunc {
+export function handleBindingClick<Res=unknown>(binding: AppBinding, context: AppContext, intl: any): ActionFunc {
+    return async (dispatch: DispatchFunc) => {
+        // Fetch form
+        if (binding.form?.source) {
+            const callRequest = createCallRequest(
+                binding.form.source,
+                context,
+            );
+
+            const res = await dispatch(doAppFetchForm<Res>(callRequest, intl, true));
+            return {data: res};
+        }
+
+        // Open form
+        if (binding.form) {
+            // This should come properly formed, but using preventive checks
+            if (!binding.form?.submit) {
+                const errMsg = intl.formatMessage({
+                    id: 'apps.error.malformed_binding',
+                    defaultMessage: 'This binding is not properly formed. Contact the App developer.',
+                });
+                return {error: makeCallErrorResponse(errMsg)};
+            }
+
+            const callRequest = createCallRequest(
+                binding.form.submit,
+                context,
+            );
+
+            dispatch(openAppsModal(binding.form, callRequest));
```

Comment:
Soon we may want to open the RHS instead of opening a modal. Currently the mobile app handles all of the "side effects" (like opening a modal) on the component side after this function has returned. Will this be difficult to refactor for that here? I don't mean in this PR

Reply:
I guess the plan is still to give the option (open in modal vs open in rhs). Either way, it should not be very different from what we have here. Regarding moving this down to the component, should be "simple" to do.